pkgs/development/tools/regclient/default.nix

```
@@ -20,22 +27,56 @@ buildGoModule rec {
   ldflags = [
     "-s"
     "-w"
-    "-X main.VCSTag=${tag}"
+    "-X github.com/regclient/regclient/internal/version.vcsTag=${tag}"
   ];
 
-  postInstall =
-    lib.concatStringsSep "\n" (
-      map (bin: ''
-        mkdir -p ''$${bin}/bin &&
-        mv $out/bin/${bin} ''$${bin}/bin/ &&
-        ln -s ''$${bin}/bin/${bin} $out/bin/
-      '') bins
-    );
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = lib.concatStringsSep "\n" (
+    map
+      (bin: ''
+        export bin=''$${bin}
+        export outputBin=bin
+
+        mkdir -p $bin/bin
+        mv $out/bin/${bin} $bin/bin
+        ln -s $bin/bin/${bin} $out/bin
+
+        $bin/bin/${bin} completion bash >${bin}.bash
+        $bin/bin/${bin} completion fish >${bin}.fish
+        $bin/bin/${bin} completion zsh >${bin}.zsh
+        installShellCompletion ${bin}.{bash,fish,zsh}
```

```suggestion
        installShellCompletion --cmd ${bin} \
          --bash <($bin/bin/${bin} completion bash) \
          --fish <($bin/bin/${bin} completion fish) \
          --zsh <($bin/bin/${bin} completion zsh)
```
we can avoid creating the files

---

It's just a bit more convenient